py/desispec/sky.py

```diff
@@ -1014,11 +1017,11 @@ def subtract_sky(frame, skymodel, apply_throughput_correction_to_lines = True, a
                         skylines = skymodel.flux[fiber] - cont
                         skylines[skylines<0]=0
                         # apply correction to the sky lines only
-                        skymodel.flux[fiber] += (skymodel.throughput_corrections[fiber]-1.)*skylines
+                        skymodel_flux[fiber] += (skymodel.throughput_corrections[fiber]-1.)*skylines
                 else :
                     log.warning("Could not determine sky continuum, do not apply throughput correction on sky lines")
 
-    frame.flux -= skymodel.flux
+    frame.flux -= skymodel_flux
     frame.ivar = util.combine_ivar(frame.ivar, skymodel.ivar)
```

Comment:
minor (and pre-dating this PR): the skymodel_flux is adjusted for throughput corrections, but the propagated ivar is not.  I think the corrections are small enough that this doesn't matter, but please confirm that you don't care (or add adjustments for ivar too just for mathematical consistency / completeness)

Reply:
yes corrections are at the level of 1% so this is negligible. I decided not to change this to save CPU and memory.